src/services/databaseservice.cpp

```diff
@@ -876,6 +876,11 @@ bool DatabaseService::setupTables() {
         version = 41;
     }
 
+    if (version < 42) {
+        queryDisk.exec(QStringLiteral("ALTER TABLE calendarItem ADD progress INTEGER DEFAULT 0"));
```

Comment:
What happens to already completed tasks with this change? You are setting the progress to "0" for those too!

Reply:
For me at least, it did not change the progress to 0 for them. The progress data is already there (completed seemed to be just checking if progress is 0 or 100). 

I have old database without any of the changes and when I use it, it updates accordingly, the completed tasks stay completed and their progress slider is just turned to 100.

Reply:
Tested it again, the priority slider is just set to max on anything that is set completed and it stays completed as it should.

Reply:
If the user doesn't open the todo dialog after a migration, the progress state of 0 could be synced to the CalDAV server. It would be more save when the migration had two more queries...

1) set the progress to 0 if completed is 0
2) set the progress to 100 if completed is 1

Reply:
Good idea. I'll add that... So something like this?

```cpp
    if (version < 42) {
        queryDisk.exec(QStringLiteral("ALTER TABLE calendarItem ADD progress INTEGER DEFAULT 0"));
        // loop over all items in calendarItem table
        // if completed is 0, set progress 0
        // if completed is 1, set progress 100
        version = 42;
    }

```

I am not very familiar with databases so how can I do this? :sweat_smile:

Reply:
It's much easier: 😉

```sql
UPDATE calendarItem SET progress = 0 WHERE completed = 0;
UPDATE calendarItem SET progress = 100 WHERE completed = 1;
```

Reply:
Thanks, I added those there. :)